JustSaying.AwsTools/QueueCreation/RegionResourceCache.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+
+namespace JustSaying.AwsTools.QueueCreation
+{
+    public class RegionResourceCache<T> : Dictionary<string, Dictionary<string, T>>, IRegionResourceCache<T>
+    {
+        public T TryGetFromCache(string region, string key)
+        {
+            if (this.ContainsKey(region) == false)
+                return default(T);
+            var regionDict = this[region];
+
+            if (regionDict.ContainsKey(key) == false)
+                return default(T);
+            return regionDict[key];
+        }
+
+        public void AddToCache(string region, string key, T value)
+        {
```

Comment:
Like most of the logic in registration of publishers and consumers, this is not thread safe. It's no issue for me as I'm only calling this in a single thread. Perhaps we should add a note in the readme about registration being non thread safe.